tests/integration/test_run_cli.py

```diff
@@ -57,3 +57,71 @@ def test_run_in_shell(pixi: Path, tmp_path: Path) -> None:
         stdout_contains=["a", "a1"],
         env=env,
     )
+
+
+def test_run_with_activation(pixi: Path, tmp_path: Path) -> None:
+    manifest = tmp_path.joinpath("pixi.toml")
+    toml = f"""
+    {EMPTY_BOILERPLATE_PROJECT}
+    [activation.env]
+    TEST_ENV_VAR_FOR_ACTIVATION_TEST = "test123"
+    [tasks]
+    task = "echo $TEST_ENV_VAR_FOR_ACTIVATION_TEST"
+    """
+    manifest.write_text(toml)
+
+    # Run the default task
+    verify_cli_command(
+        [pixi, "run", "--manifest-path", manifest, "task"],
+        ExitCode.SUCCESS,
+        stdout_contains="test123",
+    )
+
+    # Validate that without experimental it does not use the cache
+    assert not tmp_path.joinpath(".pixi/activation-env-v0").exists()
+
+    # Enable the experimental cache config
+    verify_cli_command(
+        [
+            pixi,
+            "config",
+            "set",
+            "--manifest-path",
+            manifest,
+            "--local",
+            "experimental.use-environment-activation-cache",
+            "true",
+        ],
+        ExitCode.SUCCESS,
+    )
+
+    # Run the default task and create cache
+    verify_cli_command(
+        [pixi, "run", "--manifest-path", manifest, "task"],
+        ExitCode.SUCCESS,
+        stdout_contains="test123",
+    )
+
+    # Modify the environment variable in cache
+    cache_path = tmp_path.joinpath(".pixi/activation-env-v0/activation_default.json")
+    with cache_path.open("r+") as f:
+        contents = f.read()
+        new_contents = contents.replace("test123", "test456")
+        f.seek(0)  # Move pointer to start of the file
+        f.write(new_contents)
+        f.truncate()  # Remove any remaining original content
```

Comment:
Deserializing that to a dict, doing the transformations there and serializing it again would make this more readable I think :)

Reply:
Fixed in [`e1d38c9` (#2367)](https://github.com/prefix-dev/pixi/pull/2367/commits/e1d38c9b25a99c9f4a9cd0a5349f75961b439508)

Also showed a git merge conflict :+1: